tests/python_package_test/test_sklearn.py

```diff
@@ -182,26 +177,27 @@ def test_sklearn_backward_compatibility(self):
         y_pred_2 = clf_2.fit(X_train, y_train).predict_proba(X_test)
         np.testing.assert_allclose(y_pred_1, y_pred_2)
 
+    # sklearn <0.19 cannot accept instance, but many tests could be passed only with min_data=1 and min_data_in_bin=1
+    @unittest.skipIf(not sklearn_at_least_019, 'scikit-learn version is less than 0.19')
     def test_sklearn_integration(self):
-        # sklearn <0.19 cannot accept instance, but many tests could be passed only with min_data=1 and min_data_in_bin=1
-        if sklearn_at_least_019:
-            # we cannot use `check_estimator` directly since there is no skip test mechanism
-            for name, estimator in ((lgb.sklearn.LGBMClassifier.__name__, lgb.sklearn.LGBMClassifier),
-                                    (lgb.sklearn.LGBMRegressor.__name__, lgb.sklearn.LGBMRegressor)):
-                check_parameters_default_constructible(name, estimator)
-                check_no_fit_attributes_set_in_init(name, estimator)
-                # we cannot leave default params (see https://github.com/Microsoft/LightGBM/issues/833)
-                estimator = estimator(min_child_samples=1, min_data_in_bin=1)
-                for check in _yield_all_checks(name, estimator):
-                    if check.__name__ == 'check_estimators_nan_inf':
-                        continue  # skip test because LightGBM deals with nan
-                    try:
-                        check(name, estimator)
-                    except SkipTest as message:
-                        warnings.warn(message, SkipTestWarning)
-
-    @unittest.skipIf(not IS_PANDAS_INSTALLED, 'pandas is not installed')
+        # we cannot use `check_estimator` directly since there is no skip test mechanism
+        for name, estimator in ((lgb.sklearn.LGBMClassifier.__name__, lgb.sklearn.LGBMClassifier),
+                                (lgb.sklearn.LGBMRegressor.__name__, lgb.sklearn.LGBMRegressor)):
```

Comment:
I'm afraid that there are no any built-in tests for ranking in scikit-learn. But I'll try.

Reply:
I've gave a brief glance at 
https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/estimator_checks.py
and it seems that there are still no checks for ranking.

Simple addition of `LGBMRanker` to the estimators which are tested, results in
```
ValueError: Should set group for ranking task
```
in `check_estimators_dtypes`, because sklearn doesn't pass `group` argument in their tests and from our side we cannot change anything in this case from the outside code.